src/query/functions/src/aggregates/aggregate_distinct_count.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::alloc::Layout;
+use std::fmt;
+use std::sync::Arc;
+
+use common_arrow::arrow::bitmap::Bitmap;
+use common_datavalues::prelude::*;
+use common_exception::Result;
+use common_io::prelude::*;
+use streaming_algorithms::HyperLogLog;
+
+use super::aggregate_function::AggregateFunction;
+use super::aggregate_function_factory::AggregateFunctionDescription;
+use super::StateAddr;
+use crate::aggregates::aggregator_common::assert_variadic_arguments;
+
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct AggregateCountState {
+    hll: HyperLogLog<DataValue>,
+    input_rows: usize,
+}
+
+#[derive(Clone)]
+pub struct AggregateDistinctCountFunction {
```

Comment:
We already have `AggregateDistinctCountFunction` as combinator, should rename to `ApproxCountDistinctFunction`